staging/src/k8s.io/cloud-provider/controllers/service/controller_test.go

```diff
@@ -83,6 +83,49 @@ func defaultExternalService() *v1.Service {
 	return newService("external-balancer", types.UID("123"), v1.ServiceTypeLoadBalancer)
 }
 
+func addIngressIP(s *v1.Service) *v1.Service {
```

Comment:
I know you're following the established pattern here, but I think this inside-out style negatively impacts the readability.  As an example, from below:

```
addServicePort(newService("udp-service", "", v1.ServiceTypeLoadBalancer), v1.ProtocolUDP),
```

vs
```
newService("udp-service", "", v1.ServiceTypeLoadBalancer, addPort(v1.ProtocolUDP)),
```

I would suggest something like:

First commit which changes this "inside-out" into the "tweaks" model, but doesn't change any coverage.
Second commit adds some tweak functions and changes callers to use them.

I would also suggest breaking this up into a series of smaller commits, each of which does one cleanup.  This is much easier to review.

Reply:
No problem, I will modify a version as suggested